plugins/identity-shiro/src/main/java/org/opensearch/identity/shiro/ShiroIdentityPlugin.java

```diff
@@ -138,7 +141,7 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
         }
     }
 
-    public PluginSubject getPluginSubject(Plugin plugin) {
-        return new ShiroPluginSubject(threadPool);
+    public FilterClient getRunAsClient(Plugin plugin) {
```

Comment:
I would recommend to only have the basic `Client` interface as return type. The fact that it is a filter client is just an implementation detail.